src/rapidgzip/gzip/GzipReader.hpp

```diff
@@ -118,14 +141,61 @@ class GzipReader :
             return m_currentPosition;
         }
 
+        if ( m_indexIsImported ) {
+            return m_blockMap->back().second;
+        }
+
         throw std::invalid_argument( "Can't get stream size when not finished reading at least once!" );
     }
 
     size_t
-    seek( long long int /* offset */,
-          int           /* origin */ = SEEK_SET ) final
+    seek( long long int offset ,
+          int           origin = SEEK_SET ) final
     {
-        throw std::logic_error( "Not implemented (yet)!" );
```

Comment:
Please adjust `seekable`. It was wrong from the beginning and should always have returned `false`, but now you can change it to return true in case an index was loaded (check for non-null, finalized `m_blockMap`), assuming this does not break any tests that want the wrong "seekable = true" result.